tests/baselines/reference/noUnusedLocals_writeOnly.js

```diff
@@ -3,11 +3,15 @@ function f(x = 0) {
     x = 1;
     x++;
     x /= 2;
+    ([x] = [1]);
+    ({ x } = { x: 1 });
+    ({ x: x } = { x: 1 });
```

Comment:
I think you need tests for more levels in destructuring (nested object literal)

Reply:
shouldnt each test be moved to their own function to guarantee used is getting marked correctly for each expression type usage of x?

Reply:
Every one of these tests only writes to `x`, so we're testing that *none* of these marks `x` as used. If a single test failed the baseline would change since `x` was used.